### PR TITLE
#487 Refactoring channel creation popup

### PR DESCRIPTION
**Pull request by stephanevieira75**
Capture d’écran de 2020-11-02 17-46-43

### CODE REVIEW

File: twake/frontend/src/app/scenes/Client/ChannelsBar/ChannelTemplateEditor.tsx [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+  useEffect(() => {
+    onChange({
+      icon,
+      name,
+      description,
+      visibility,
+      // TODO find a better way to get company id
+      company_id: OldCollections.get('workspaces').find(workspaceId)?.group?.id,
+      workspace_id: workspaceId,
```
Comment (RomaricMourgues) [inline twake/frontend/src/app/scenes/Client/ChannelsBar/ChannelTemplateEditor.tsx:38]: I don't want this to come from the template editor, please move this code to twake/frontend/src/app/scenes/Client/ChannelsBar/ChannelWorkspaceEditor.tsx

File: twake/frontend/src/app/scenes/Client/ChannelsBar/ChannelTemplateEditor.tsx [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    });
+  }, [icon, name, description, visibility]);
```
Comment (RomaricMourgues) [inline twake/frontend/src/app/scenes/Client/ChannelsBar/ChannelTemplateEditor.tsx:40]: Maybe we'll find a better way to do this :)

File: twake/frontend/src/app/scenes/Client/ChannelsBar/ChannelWorkspaceEditor.tsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    company_id: '',
+    workspace_id: '',
+    visibility: 'private',
+  });
+
+  const collectionPath: string = `/companies/${newChannel.company_id}/workspaces/${newChannel.workspace_id}/channels/`;
+  const ChannelsCollections = Collections.get(collectionPath);
+
+  const onChange = (channelEntries: ChannelType): void => {
+    return setNewChannel(Object.assign({}, channelEntries));
```
Comment (RomaricMourgues) [inline twake/frontend/src/app/scenes/Client/ChannelsBar/ChannelWorkspaceEditor.tsx:28]: We don't need to use a state for this, do we show the channel components in this component ? No we don't.
So you can use a simple variable to store the template object (all in one variable by the way).

**Review by RomaricMourgues — CHANGES_REQUESTED** [review]
Just very small changes, good work!

File: twake/frontend/src/app/scenes/Client/ChannelsBar/ChannelWorkspaceEditor.tsx (hunk 3 above)
Comment (stephanevieira75) [reply twake/frontend/src/app/scenes/Client/ChannelsBar/ChannelWorkspaceEditor.tsx]: fixed :+1:

File: twake/frontend/src/app/scenes/Client/ChannelsBar/ChannelTemplateEditor.tsx (hunk 1 above)
Comment (stephanevieira75) [reply twake/frontend/src/app/scenes/Client/ChannelsBar/ChannelTemplateEditor.tsx]: done

**Pull request merged**